unittests/test_schedulers.py

```diff
@@ -544,116 +579,257 @@ def setUp(self):
-        self.testjob._sched_partition = 'Invalid'
-        self.assertRaises(JobError, self.prepare_job)
+    def test_not_valid_partition_short(self):
+        self.testjob.options = ['-p invalid']
+        with self.assertRaises(JobError):
+            self.prepare_job()
+
+    def test_not_valid_partition_long(self):
+        self.testjob.options = ['--partition=invalid']
+        with self.assertRaises(JobError):
+            self.prepare_job()
 
     def test_not_valid_constraint(self):
-        self.testjob._sched_reservation = 'Foo'
         self.testjob.options = ['--constraint=invalid']
-        self.assertRaises(JobError, self.prepare_job)
+        with self.assertRaises(JobError):
+            self.prepare_job()
+
+    def test_valid_reservation_cmd(self):
+        self.testjob._sched_access = ['--constraint=f2']
+        self.testjob._sched_reservation = 'dummy'
+        self.testjob._get_reservation_nodes = self.create_reservation_nodes
+        self.prepare_job()
+        self.assertEqual(self.testjob.num_tasks, 4)
 
-    def test_noreservation(self):
-        self.assertRaises(JobError, self.prepare_job)
+    def test_valid_reservation_option(self):
+        self.testjob._sched_access = ['--constraint=f2']
+        self.testjob.options = ['--reservation=dummy']
+        self.testjob._get_reservation_nodes = self.create_reservation_nodes
+        self.prepare_job()
+        self.assertEqual(self.testjob.num_tasks, 4)
+
+    def test_exclude_nodes_cmd(self):
+        self.testjob._sched_access = ['--constraint=f1']
+        self.testjob._sched_exclude_nodelist = 'nid00001'
+        self.testjob._get_nodes_by_name = self.get_nodes_by_name
+        self.prepare_job()
+        self.assertEqual(self.testjob.num_tasks, 8)
+
+    def test_exclude_nodes_short(self):
+        self.testjob._sched_access = ['--constraint=f1']
+        self.testjob.options = ['-x nid00001']
+        self.testjob._get_nodes_by_name = self.get_nodes_by_name
+        self.prepare_job()
+        self.assertEqual(self.testjob.num_tasks, 8)
+
+    def test_exclude_nodes_long(self):
+        self.testjob._sched_access = ['--constraint=f1']
+        self.testjob.options = ['--exclude=nid00001']
+        self.testjob._get_nodes_by_name = self.get_nodes_by_name
+        self.prepare_job()
+        self.assertEqual(self.testjob.num_tasks, 8)
 
     def prepare_job(self):
         self.testjob.prepare(['hostname'])
 
 
-class TestSlurmFlexibleNodeAllocationExclude(TestSlurmFlexibleNodeAllocation):
-    def create_dummy_exclude_nodes(obj):
-        return [obj.create_dummy_nodes()[0].name]
-
+class TestSlurmNode(unittest.TestCase):
     def setUp(self):
-        super().setUp()
-        self.testjob._sched_exclude_nodelist = 'nid00001'
-        # monkey patch `_get_exclude_nodes` to simulate extraction of
-        # slurm nodes through the use of `scontrol show`
-        self.testjob._get_excluded_node_names = self.create_dummy_exclude_nodes
+        node_description = (
```

Comment:
Why do you need this? Aren't the rest of the node description examples enough?

Reply:
Applied the suggested change